pkg/dynamicaccess/history_test.go

```diff
@@ -1,59 +1,111 @@
-		{input: now.AddDate(-2, -0, 0).Unix(), expected: "value2"},
-		{input: now.AddDate(-2, -6, 0).Unix(), expected: "value1"},
-		{input: now.AddDate(-3, -0, 0).Unix(), expected: "value1"},
-	}
+func TestHistoryAdd(t *testing.T) {
+	h, err := dynamicaccess.NewHistory(nil, nil)
+	assert.NoError(t, err)
 
-	for _, tt := range tests {
-		t.Run("", func(t *testing.T) {
-			sAt, _ := h.Lookup(tt.input)
-			output, _ := sAt.Get([]byte("key1"))
-			assert.Equal(t, output, hex.EncodeToString([]byte(tt.expected)))
-		})
-	}
+	addr := swarm.NewAddress([]byte("addr"))
+
+	ctx := context.Background()
+
+	err = h.Add(ctx, addr, nil)
+	assert.NoError(t, err)
 }
 
-func prepareTestHistory() dynamicaccess.History {
-	var (
-		h   = mock.NewHistory()
-		now = time.Now()
-		s1  = kvsmock.New()
-		s2  = kvsmock.New()
-		s3  = kvsmock.New()
-	)
-	s1.Put([]byte("key1"), []byte("value1"))
-	s2.Put([]byte("key1"), []byte("value2"))
-	s3.Put([]byte("key1"), []byte("value3"))
-
-	h.Insert(now.AddDate(-3, 0, 0).Unix(), s1)
-	h.Insert(now.AddDate(-2, 0, 0).Unix(), s2)
-	h.Insert(now.AddDate(-1, 0, 0).Unix(), s3)
-
-	return h
+func TestSingleNodeHistoryLookup(t *testing.T) {
+	storer := mockstorer.New()
+	ctx := context.Background()
+	ls := loadsave.New(storer.ChunkStore(), storer.Cache(), pipelineFactory(storer.Cache(), false))
+
+	h, err := dynamicaccess.NewHistory(ls, nil)
+	assert.NoError(t, err)
+
+	testActRef := swarm.RandAddress(t)
+	err = h.Add(ctx, testActRef, nil)
+	assert.NoError(t, err)
+
+	_, err = h.Store(ctx)
+	assert.NoError(t, err)
+
+	searchedTime := time.Now().Unix()
+	actRef, err := h.Lookup(ctx, searchedTime, ls)
+	assert.NoError(t, err)
+	assert.True(t, actRef.Equal(testActRef))
+}
+
+func TestMultiNodeHistoryLookup(t *testing.T) {
+	storer := mockstorer.New()
+	ctx := context.Background()
+	ls := loadsave.New(storer.ChunkStore(), storer.Cache(), pipelineFactory(storer.Cache(), false))
+
+	h, _ := dynamicaccess.NewHistory(ls, nil)
+
+	testActRef1 := swarm.NewAddress([]byte("39a5ea87b141fe44aa609c3327ecd891"))
+	firstTime := time.Date(1994, time.April, 1, 0, 0, 0, 0, time.UTC).Unix()
+	h.Add(ctx, testActRef1, &firstTime)
+
+	testActRef2 := swarm.NewAddress([]byte("39a5ea87b141fe44aa609c3327ecd892"))
+	secondTime := time.Date(2000, time.April, 1, 0, 0, 0, 0, time.UTC).Unix()
+	h.Add(ctx, testActRef2, &secondTime)
+
+	testActRef3 := swarm.NewAddress([]byte("39a5ea87b141fe44aa609c3327ecd893"))
+	thirdTime := time.Date(2015, time.April, 1, 0, 0, 0, 0, time.UTC).Unix()
+	h.Add(ctx, testActRef3, &thirdTime)
+
+	testActRef4 := swarm.NewAddress([]byte("39a5ea87b141fe44aa609c3327ecd894"))
+	fourthTime := time.Date(2020, time.April, 1, 0, 0, 0, 0, time.UTC).Unix()
+	h.Add(ctx, testActRef4, &fourthTime)
+
+	testActRef5 := swarm.NewAddress([]byte("39a5ea87b141fe44aa609c3327ecd895"))
+	fifthTime := time.Date(2030, time.April, 1, 0, 0, 0, 0, time.UTC).Unix()
+	h.Add(ctx, testActRef5, &fifthTime)
+
+	_, err := h.Store(ctx)
```

Comment:
Store() should be tested separately, e.g.: create a histroy and store it, then construct another history with the ref returned by Store.

Reply:
Is it necessary to retest a tested fn?

`func (h *history) Store(ctx context.Context) (swarm.Address, error) {
	return h.manifest.Store(ctx)
}
`

It's just a wrapper.